app/uk/gov/hmrc/cdsimportsddsfrontend/services/xml/InvoiceLineXmlWriter.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsimportsddsfrontend.services.xml
+
+import cats.implicits._
+import uk.gov.hmrc.cdsimportsddsfrontend.domain.InvoiceLine
+import uk.gov.hmrc.cdsimportsddsfrontend.services.xml.XmlWriterInstances.itemChargeAmountWriter
+
+import scala.xml.{Elem, Node}
+
+object InvoiceLineXmlWriter {
+
+  implicit val invoiceLineXmlWriter: XmlWriter[InvoiceLine] = new XmlWriter[InvoiceLine] {
+    override def toXmlOption(value: InvoiceLine): Option[Elem] = {
+      val itemChargeAmount = value.itemChargeAmount.flatMap(itemChargeAmountWriter.toXmlOption)
+
+      val nodes: List[Node] = List(itemChargeAmount).flattenOption
```

Comment:
similarly, this could possibly be simplified?

Reply:
Done